doc/sphinx/source/frontend/api/Settings/eqsys/n_re.rst

```diff
@@ -556,6 +573,57 @@ An example of how the mode for the critical effective field can be set to ``CYLI
    ds.eqsys.n_re.setEceff(RunawayElectrons.COLLQTY_ECEFF_MODE_CYLINDRICAL)
 
 
+Negative electric fields
+------------------------
+Runaway generation is generally treated correctly regardless of the sign of the
+sign of the electric field in DREAM. One exception occurs for the avalanche
+source when the electric field *changes* sign during the simulation. In this
+case, in default mode, the sign change may cause excessive avalanching in the
+new direction of the electric field. To resolve this issue, the density of
+runaways travelling anti-parallel to the magnetic field (called `n_re_neg` in
+the code) can be introduced in the system. When this is done, the kinetic
+avalanche source is automatically modified to account for whether runaways are
+travelling parallel or anti-parallel to the magnetic field.
+
+.. note::
+
+   Tracking the direction of motion of the runaways requires the runaway grid
+   to enabled.
+
+The density of runaway electrons travelling anti-parallel to the magnetic field
+is defined as
+
+.. math::
+
+   \left\langle n_{\rm re}^{\rm neg}\right\rangle = 
+       \frac{1}{V'}\int_{p_{\rm re}}^{-\infty} \int_{-1}^1 f_{\rm re}
+       \mathcal{V}'\,\mathrm{d}\xi_0 \mathrm{d} p.
+
+The avalanche source term in the kinetic equation is then modified to become
+
+.. math::
+
+   S(n_{\rm re}) \to S^{+}(n_{\rm re}) - S^{+}(n_{\rm re}^{\rm neg}) + S^{-}(n_{\rm re}^{\rm neg})
+
+where the superscript sign indicates whether the source creates electrons in the
+parallel (+) or anti-parallel (-) direction (corresponding to positive and
+negative values of :math:`\xi_0` respectively). In the default source, the
+avalanche electrons are created in the direction instantaneously parallel to the
+electric field.
+
+Example
+^^^^^^^
+To include the density of anti-parallel runaways, simply do the following:
+
+.. code-block::
+
+   from DREAM import DREAMSettings
+
+   ds = DREAMSettings()
+   ...
+   ds.eqsys.n_re.setNegativeRunaways(True)
```

Comment:
Since n_re_neg is computed as a moment of f_re (and thus adds as many elements to the matrix as f_re consists of), including it will add alot of non-zero elements. For the usual n_re it's different, since it's evolved via the flux of particles into the runaway grid, and hence will at most add O(nXi_hot*nr) elements to the matrix.

Also, n_re_neg is only needed in scenarios where the electric field changes sign during the evolution. If the electric field remains active in the same direction throughout the simulation, the avalanche source is evaluated correctly (regardless of which direction the electric field points in).

Hence, to allow for some optimization, I decided to make this an explicit switch.

Reply:
Oh, I think I missed the second part of your comment...

In cases where E changes sign, we actually want to keep the avalanche term active on both sides, since it will lead to slowing down on one side (well, kind of...), and acceleration on the other. It might be possible to add some logic to the term to disable itself before the E field has changed sign, however the operators acting on ``n_re_neg`` are already automatically disabled when ``n_re_neg`` is zero (since we don't insert elements which are identically zero into the matrix). We might save some time by disabling the operator acting on ``n_re``, but it's already way cheaper to use since ``n_re`` is *not* a moment of ``f_re``. We could maybe also save some time in building the operators, but considering that building these operators is probably a miniscule part of the entire matrix/vector construction stage I'm not sure it's worth the effort.